client/pages/product-details.jsx

```diff
@@ -0,0 +1,161 @@
+import React from 'react';
+import Container from 'react-bootstrap/Container';
+import parseRoute from '../lib/parse-route';
+import Row from 'react-bootstrap/Row';
+import ListGroup from 'react-bootstrap/ListGroup';
+import Accordion from 'react-bootstrap/Accordion';
+
+export default class ProductDetails extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      product: {},
+      route: parseRoute(window.location.hash),
+      loading: false
```

Comment:
We use `loading: true` initially for the purpose of showing a loading spinner or some kind of placeholder until the data arrives.